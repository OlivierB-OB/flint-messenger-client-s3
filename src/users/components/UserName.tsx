import React, { Fragment } from 'react';
import { IUserInfo } from '../types';

export interface IUserNameProps {
  info?: IUserInfo;
}

export function UserName({ info }: IUserNameProps) {
  const { lastName, firstName } = info || { firstName: 'Unknown', lastName: 'user' };
  const details = `${firstName} ${lastName}`;
  return <Fragment>{details}</Fragment>;
}
