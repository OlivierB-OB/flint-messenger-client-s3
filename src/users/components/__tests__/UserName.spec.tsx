import React from 'react';
import { shallow } from 'enzyme';
import { UserName } from '../UserName';

describe('UserName', () => {
  it('should display the user name', async () => {
    const info: any = {
      firstName: 'foo',
      lastName: 'bar',
    };
    const component = shallow(<UserName info={info} />);
    expect(component).toMatchSnapshot();
  });

  it('should support undefined user information', async () => {
    const component = shallow(<UserName info={undefined} />);
    expect(component).toMatchSnapshot();
  });
});
