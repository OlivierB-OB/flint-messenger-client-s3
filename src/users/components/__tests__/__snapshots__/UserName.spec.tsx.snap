// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserName should display the user name 1`] = `
<Fragment>
  foo bar
</Fragment>
`;

exports[`UserName should support undefined user information 1`] = `
<Fragment>
  Unknown user
</Fragment>
`;
